app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt

```diff
@@ -185,6 +189,40 @@ class ThreadViewModel @Inject constructor(
         }
     }
 
+    fun fetchCalendarEvents(messages: List<Message>) {
+        fetchCalendarEvent?.cancel()
+        fetchCalendarEvent = viewModelScope.launch(ioCoroutineContext) {
+            mailboxContentRealm().writeBlocking {
+                messages.forEach { message ->
+                    if (!message.isFullyDownloaded()) return@forEach // Only treat message that have their attachments downloaded
+
+                    val alreadyTreated = !treatedMessagesForCalendarEvent.add(message.uid)
+                    if (alreadyTreated) return@forEach
+
+                    val icsAttachments = message.attachments.filter { it.mimeType == "application/ics" }
+                    if (icsAttachments.count() != 1) return@forEach
+
+                    val icsAttachment = icsAttachments.single()
+
+                    val calendarEventResponse = icsAttachment.resource?.let { resource ->
+                        ApiRepository.getAttachmentCalendarEvent(resource)
+                    } ?: return@forEach
+
+                    if (calendarEventResponse.isSuccess()) {
+                        messageController.updateCalendarEvent(message.uid, calendarEventResponse.data!!, realm = this)
+                    } else {
+                        Sentry.withScope { scope ->
+                            scope.setExtra("ics attachment mimeType", icsAttachment.mimeType)
+                            scope.setExtra("ics attachment size", icsAttachment.size.toString())
+                            scope.setExtra("error code", calendarEventResponse.error?.code.toString())
+                            Sentry.captureMessage("Failed loading calendar event")
+                        }
+                    }
+                }
```

Comment:
I don't know to what extent this actually is an issue. But anyway, we could always have a `fetchEventCalendarEvent(message: Message)` that does all of the logic for a single message which would be called inside the for loop, do you like that idea?

In any case this bit of code changes a lot and moves around in future PRs so changing it right now would cause a lot of merge conflicts, maybe we can at least do this modification at the end or something

Reply:
That's what I had in mind, so no problem with me